contrib/hashicorp/consul/consul_test.go

```diff
@@ -0,0 +1,249 @@
+package consul
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	consul "github.com/hashicorp/consul/api"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMain(m *testing.M) {
+	_, ok := os.LookupEnv("INTEGRATION")
+	if !ok {
+		fmt.Println("--- SKIP: to enable integration test, set the INTEGRATION environment variable")
+		os.Exit(0)
+	}
+	os.Exit(m.Run())
+}
+
+func kv() *KV {
+	client, err := NewClient(consul.DefaultConfig())
+	if err != nil {
+		panic(err)
+	}
+	// Get a handle to the KV API
+	kv := client.KV()
+	return kv
+}
+
+func TestClient(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	client, err := NewClient(consul.DefaultConfig())
+	if err != nil {
+		panic(err)
+	}
+
+	assert.NotNil(client)
+	assert.Nil(err)
+}
+
+func TestClient_error(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	var config = &consul.Config{
+		Address: "badscheme://baduri",
+	}
+
+	client, err := NewClient(config)
+	assert.Nil(client)
+	assert.Error(err)
+
+}
+
+func TestClient_KV(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	client, _ := NewClient(consul.DefaultConfig())
+	clientCtx := client.ctx
+	kv := client.KV()
+	kvCtx := kv.ctx
+
+	assert.Equal(clientCtx, kvCtx)
+}
+
+func TestKV_Put(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+	kv := kv()
+
+	p := &consul.KVPair{Key: "test_key", Value: []byte("test_value")}
+	kv.Put(p, nil)
+
+	spans := mt.FinishedSpans()
+	assert.Len(spans, 1)
+	span := spans[0]
+	assert.Equal("consul.command", span.OperationName())
+	assert.Equal("PUT", span.Tag(ext.ResourceName))
+	assert.Equal(ext.SpanTypeConsul, span.Tag(ext.SpanType))
+	assert.Equal("consul", span.Tag(ext.ServiceName))
+}
+
+func TestKV_Get(t *testing.T) {
```

Comment:
There's a lot of repetition of test setup here.
Subtests could be clearer and less repetitive.